crates/source_analyzer/src/contexts.rs

```diff
@@ -433,571 +350,16 @@ fn extract_three_arguments(
                 ))) => Ok((
                     type_argument.clone(),
                     first_literal_argument.token().to_string().trim_matches('"').to_string(),
-                    second_literal_argument.token().to_string().trim_matches('"').split('.').map(PathSegment::from).collect(),
+                    Path::from(second_literal_argument.token().to_string().trim_matches('"')),
                 )),
-                _ => new_syn_error_as_eyre_result(
-                    arguments.span(),
-                    "expected type in first generic parameter and string literals in second and third generic parameters",
-                    file_path,
+                _ => Err(
+                    ParseError::new_spanned(&arguments.args,"expected type in first generic parameter and string literals in second and third generic parameters")
                 ),
             }
         }
-        _ => new_syn_error_as_eyre_result(
-            arguments.span(),
+        _ => Err(ParseError::new_spanned(
+            arguments,
             "expected exactly three generic parameters",
-            file_path,
-        ),
-    }
-}
-
-#[cfg(test)]
```

Comment:
They are gone and should be added if there is no reason to remove them. Maybe they need to be written more cleanly. But messy tests are at least better than removed tests.

Reply:
Readded the tests.

Removed the one regarding cartesian products when having multiple variables in a path since that feature was apparently removed.
Modified the variable expansion function interface on the `Path` type slightly. It doesn't need a whole cycler, just the instances.
Readded a check for AdditionalOutput paths containing optionals to make one of the tests pass again.

Reply:
I now noticed that this multi-variable feature is only partially removed and half-done. We should rethink what the goal is:

For now we do not need to support additional variables because I currently see no use-case where we need them (so it is okay that Schmiddy dropped this feature). This may change in the future, but YAGNI.

In addition, variables restrict the scope in a way because they allow conditional branches in the type path. Once a branch is taken by expanding the variable, subsequent branches of the same variable in deeper path segments become obsolete. We therefore only need to support a single occurrence of any variable in the path. In combination that we only need to support cycler instances leads to a single cycler instance per path.

Please adapt the `Path` type s.t. it fits this design decisions best. IMHO, we should also enforce that the cycler instance variable is named `cycler_instance` to communicate a consistent style for our users.

The current implementation is insufficient, because it allows you to write any variable name if you have a `$` character in front. It will still be replaced with cycler instances. In addition, what happens if we have multiple variables in the path? This should also be not possible as discussed above.

Reply:
The `Path` implementation now checks for (dis)allowed optional segments, no more than one variable segment as well as the names of variable segments.